src/bundle/Controller/TabController.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * File containing the TabController class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\HybridPlatformUiBundle\Controller;
+
+use eZ\Publish\API\Repository\ContentService;
+use eZ\Publish\Core\MVC\Symfony\Controller\Controller;
+use EzSystems\HybridPlatformUi\Response\ResetResponse;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\Routing\RouterInterface;
+
+abstract class TabController extends Controller
+{
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var ContentService
+     */
+    private $contentService;
+
+    public function __construct(
+        RouterInterface $router,
+        ContentService $contentService
+    ) {
+        $this->router = $router;
+        $this->contentService = $contentService;
+    }
+
+    protected function resetLocation($contentId)
+    {
+        $mainLocationId = $this->contentService->loadContentInfo($contentId)->mainLocationId;
+
+        return new ResetResponse($this->generateTabUrl($contentId, $mainLocationId));
+    }
+
+    protected function redirectToLocationsTab($contentId, $locationId)
```

Comment:
I wouldn't do that.

Tabs are meant to be extensible in the end, this won't be usable. Instead, I'd assume that we standardize the name for the tabs views (`xxx_tab`). We can pass the tab identifier to the method, and `_tab` will be added automatically:

```
$this->redirectToTab('locations', $contentId, $locationId);
```

What bothers me a bit more is that in that case, it is NOT a redirection, from the hybrid-platform-ui perspective. The protocol we have established makes that a refresh, hence my suggestion in the story.